src/runtime/runtime_atomics.h

```diff
@@ -0,0 +1,270 @@
+#ifndef HALIDE_RUNTIME_RUNTIME_ATOMICS_H
+#define HALIDE_RUNTIME_RUNTIME_ATOMICS_H
+
```

Comment:
It is likely worth capturing the issue re: __atomic possibly leading to libatomic references and that being the reason for this support. (I.e. that in order for __atomic to generate hardware instructions rather than calls to libatomic routines, all operations promised by __atomic for the datatype in question must be supported in hardware)

Reply:
As in, open an Issue and reference it in this file? Will do.

Reply:
PTAL